packages/venia-ui/lib/drivers/adapter.js

```diff
@@ -86,15 +90,93 @@ const VeniaAdapter = props => {
         return null;
     }
 
+    let storeCodeRouteHandler = '';
+    const browserRouterProps = {};
+    if (process.env.USE_STORE_CODE_IN_URL === 'true') {
+        const storeCode = storage.getItem('store_view_code') || STORE_VIEW_CODE;
+        browserRouterProps.basename = `/${storeCode}`;
+        const storeCodes = [];
+        const storeCurrencies = {};
+        if (Array.isArray(AVAILABLE_STORE_VIEWS)) {
+            AVAILABLE_STORE_VIEWS.forEach(store => {
+                storeCodes.push(store.code);
+                storeCurrencies[store.code] =
+                    store.default_display_currency_code;
+            });
+        }
+
+        /**
+         * The current store code won't be matched as it's included as the basename, if we match another store
+         * we need to change the current store.
+         */
+        storeCodeRouteHandler = (
+            <Switch>
+                <Route path={`/:storeCode(${storeCodes.join('|')})?`}>
+                    {({ match }) => {
+                        if (
+                            match.params.storeCode &&
+                            storeCodes.includes(match.params.storeCode)
+                        ) {
+                            /**
+                             * Only execute if one store code is present in the URL, multiple store codes will cause
+                             * the store state to break and cause weird side effects for the user
+                             */
+                            const regex = new RegExp(
+                                `(${storeCodes.join('|')})\/`,
+                                'g'
+                            );
+                            const storeCodesInUrl = window.location.pathname.match(
+                                regex
+                            ).length;
+                            if (storeCodesInUrl === 1) {
+                                storage.setItem(
+                                    'store_view_code',
+                                    match.params.storeCode
+                                );
+                                storage.setItem(
+                                    'store_view_currency',
+                                    storeCurrencies[match.params.storeCode]
+                                );
+                            } else {
+                                console.warn(
+                                    'Multiple store codes present in URL.'
+                                );
+                            }
+                        }
```

Comment:
I'm seeing a warning when navigating. Looks like you need to return something from this block.

![](https://i.gyazo.com/37149fbe1cfa469b4a3fd4df1c875f24.png)

Reply:
Good catch, fixed.